src/main/java/com/cdancy/bitbucket/rest/domain/pullrequest/Reference.java

```diff
@@ -31,11 +31,14 @@
 
     public abstract MinimalRepository repository();
 
+    @Nullable
+    public abstract String displayId();
+
     Reference() {
     }
 
-    @SerializedNames({ "id", "repository" })
-    public static Reference create(String id, MinimalRepository repository) {
-        return new AutoValue_Reference(id != null ? id : "refs/heads/master", repository);
+    @SerializedNames({"id", "repository", "displayId"})
+    public static Reference create(String id, MinimalRepository repository, String displayId) {
```

Comment:
Ok. Then maybe check for non-null and if so split on `/` and use last index? Something like this might work.

```
public static Reference create(String id, MinimalRepository repository) { 
    String displayId = null;
    if (id != null) {
        String parts [] = id.split("/");
        displayId = parts[parts.length];
    }
    return create(id, repository, displayId);
}
```